vm/src/builtins/set.rs

```diff
@@ -649,49 +652,103 @@ impl PySet {
 
     #[pymethod]
     fn intersection_update(
-        &self,
-        others: PosArgs<ArgIterable>,
+        zelf: PyRef<Self>,
+        others: PosArgs<PyObjectRef>,
         vm: &VirtualMachine,
     ) -> PyResult<()> {
-        self.inner.intersection_update(others.into_iter(), vm)?;
+        let arguments = others.into_vec();
+        if arguments.len() == 1 {
+            if let Some(iterable) = arguments.first() {
+                if zelf.is(iterable) {
+                    return Ok(());
+                }
+            }
```

Comment:
I think this check is not only adaptable for `len() == 1` case but also do for every argument.

For example, how about `s = set(['a']); s.intersection_update(s, s, s, s, s, s)`?